pkg/cli/build.go

```diff
@@ -108,14 +115,23 @@ func addDockerfileFlag(cmd *cobra.Command) {
 
 func addUseCogBaseImageFlag(cmd *cobra.Command) {
 	cmd.Flags().BoolVar(&buildUseCogBaseImage, "use-cog-base-image", false, "Use pre-built Cog base image for faster cold boots")
-	cmd.Flags().VisitAll(func(f *pflag.Flag) {
-		if f.Name == "use-cog-base-image" {
-			f.Hidden = true
-		}
-	})
 }
 
 func addBuildTimestampFlag(cmd *cobra.Command) {
 	cmd.Flags().Int64Var(&config.BuildSourceEpochTimestamp, "timestamp", -1, "Number of seconds sing Epoch to use for the build timestamp; this rewrites the timestamp of each layer. Useful for reproducibility. (`-1` to disable timestamp rewrites)")
 	_ = cmd.Flags().MarkHidden("timestamp")
 }
+
+func checkMutuallyExclusiveFlags(cmd *cobra.Command) error {
+	flags := []string{"use-cog-base-image", "use-cuda-base-image", "dockerfile"}
+	var flagsSet []string
+	for _, flag := range flags {
+		if cmd.Flag(flag).Changed {
+			flagsSet = append(flagsSet, "--"+flag)
+		}
+	}
+	if len(flagsSet) > 1 {
+		return fmt.Errorf("the flags %s are mutually exclusive, you can only set one of them", strings.Join(flagsSet, " and "))
```

Comment:
```suggestion
		return fmt.Errorf("The flags %s are mutually exclusive: you can only set one of them.", strings.Join(flagsSet, " and "))
```

Little bit of formatting to make it match other `fmt.Errorf` invocations.